src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -491,6 +493,26 @@ public function validateSchema(\DOMDocument $dom)
     }
 
     /**
+     * Validates an alias.
+     *
+     * @param \DOMElement
+     */
+    private function validateAlias(\DOMElement $alias, $file)
+    {
+        foreach (array('class', 'shared', 'synthetic', 'lazy', 'abstract', 'parent', 'decorates', 'decoration-inner-name', 'decoration-priority', 'autowire') as $key) {
```

Comment:
instead of hardcoding a blacklist, which will need to be updated each time we add a new feature (and will be forgotten), couldn't we use a whitelist instead, as we only allow 3 attributes (``id``, ``alias`` and ``public``) ?